src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -832,7 +832,11 @@ void ShenandoahHeap::notify_heap_changed() {
   // Update monitoring counters when we took a new region. This amortizes the
   // update costs on slow path.
   monitoring_support()->notify_heap_changed();
-  control_thread()->notify_heap_changed();
+
+  // This is called from allocation path, and thus should be fast.
+  if (_heap_changed.is_set()) {
```

Comment:
Um. The old code tests `is_unset`, then does `set`. This one introduces a bug: it would be set when already set. Should maybe be just `_heap_changed.try_set()`?

Reply:
Yikes. Good catch. I originally had `try_set` here, but then wanted to make it the same as the original code and wrote a typo! I'll go with `try_set`.